support/cas-server-support-ldap/src/main/java/org/apereo/cas/authentication/LdapAuthenticationHandler.java

```diff
@@ -101,6 +105,18 @@ public LdapAuthenticationHandler(final String name, final ServicesManager servic
         this.passwordPolicyHandlingStrategy = strategy;
     }
 
+    @Override
+    public void destroy() {
+        val authenticationHandler = (AbstractAuthenticationHandler) authenticator.getAuthenticationHandler();
+        val dnResolver = (SearchDnResolver) authenticator.getDnResolver();
+        val entryResolver = (AbstractSearchEntryResolver) authenticator.getEntryResolver();
+        authenticationHandler.getConnectionFactory().close();
+        dnResolver.getConnectionFactory().close();
+        if (entryResolver != null) {
+            entryResolver.getConnectionFactory().close();
+        }
+    }
```

Comment:
No. There are three independent thread pools. as I figured out from `LdapUtils`.

Take this as example
https://github.com/apereo/cas/blob/master/support/cas-server-support-ldap-core/src/main/java/org/apereo/cas/util/LdapUtils.java#L607

and follow the method `getAuthenticatedOrAnonSearchAuthenticator()`, it indeed creates three `newLdaptiveConnectionFactory` instances. So three `close()` is needed.

Reply:
In that case, all of this code should be moved over to ldaptive. We should not have to travel down the hierarchy to close everything. This is not maintainable. `authenticator.close() ` should be enough, and the rest should be handled by ldaptive which would know what needs closing and how and in what order, etc.

Reply:
Pinging @dfish3r; What do you think?

Reply:
I guess this is intended. As you asked before, it is possible to shared same `ConnectionFactory` to multiple objects.

Reply:
> Pinging @dfish3r; What do you think?

This is such a common problem that I've added a `close` implementation in the latest snapshot.
https://github.com/vt-middleware/ldaptive/blob/master/core/src/main/java/org/ldaptive/auth/Authenticator.java#L266
Since authenticator components could be custom, close isn't a guarantee. However, if you're using supplied implementations it will work as desired. I imagine that's 99% of use cases so it's a method worth having.

Reply:
So I would need to wait for a new snapshot version update to change these to authenticator.close()

Reply:
> authenticator.close() should be enough.

BTW I actually misread the this comment before. I agree to use authenticator.close() if there is one.